pkg/controller/kieapp/defaults/merge.go

```diff
@@ -131,42 +132,42 @@ func getRoleBindingReferenceSlice(objects []rbacv1.RoleBinding) []v1.OpenShiftOb
 func mergeDeploymentConfigs(baseline []appsv1.DeploymentConfig, overwrite []appsv1.DeploymentConfig) []appsv1.DeploymentConfig {
 	if len(overwrite) == 0 {
 		return baseline
-	} else if len(baseline) == 0 {
+	}
+	if len(baseline) == 0 {
 		return overwrite
-	} else {
-		baselineRefs := getDeploymentConfigReferenceSlice(baseline)
-		overwriteRefs := getDeploymentConfigReferenceSlice(overwrite)
-		for overwriteIndex := range overwrite {
-			overwriteItem := overwrite[overwriteIndex]
```

Comment:
@ruromero This is actually the issue that I previously tried to explain to you, that caused a bug elsewhere. Removing it introduces the same bug in this block. Your overwriteItem, which is the second return value of the range loop, is a copy and not the actual array element. Changes to it are not reflected in the overwrite array.

This introduces a second issue, which is that by trying to prove this with a test case, I realized much (if not all) of this code block is unnecessary. Mergo clearly has restrictions for what we are trying to do, but its restrictions are not as wide as I had assumed. It appears for example, that the mergo.merge of line 375 traverses a few levels deep into at least the content of ObjectMeta. Can you validate, and remove unnecessary code?